### PR TITLE
[OY2-19808] Create Textarea Component

### PR DESCRIPTION
**Pull request by jphechter**
## Purpose

Create the TextArea component.

## Type of Request

feature

## Approach

This is essentially a clone of the TextInput with some additional logic for a character counter.

#### Pull Request Creator Checklist

- [X] It is documented what type of request this is.
- [X] There is a detailed description for the purpose of this PR.
- [X] This PR is in line with the intention of this Project

### CODE REVIEW

File: src/components/TextArea/TextArea.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  placeholder: "Placeholder Text",
+  inputSuccess: true,
+};
+
+InputFilter.args = {
+  fieldName: "input-type-text",
+  label: "This field only accepts a numerical input",
+  placeholder: "Numbers only",
+  inputFilter: /^-?\d*$/i,
+};
```
Comment (jdinh8124) [inline src/components/TextArea/TextArea.stories.tsx:107]: Regarding the comment above, I believe using the args object to set the default values would prevent us from having to add repeating properties down the road like fieldName for our examples.

**Review by jdinh8124 — COMMENTED** [review]
Some comments, everything else looks good though

File: src/components/TextArea/TextArea.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    },
+    placeholder: {
+      defaultValue: undefined,
+      description: "Input field placeholder text.",
+    },
+    required: {
+      defaultValue: false,
+      description:
+        "Adds semantic required attr and appends an * to the end of the input label.",
+    },
```
[the remark this answers is not in the document]
Comment (jphechter) [reply src/components/TextArea/TextArea.stories.tsx:56]: This is a good callout, but I think we're already using the new pattern according to [the storybook  docs](https://storybook.js.org/docs/react/api/argtypes). This deprecation is under a section labeled `From version 6.2.x to 6.3.0` and we're running `6.5.9`.

File: src/components/TextArea/TextArea.stories.tsx (hunk 1 above)
Comment (jphechter) [reply src/components/TextArea/TextArea.stories.tsx:107]: I thought `defaultValue` was the default value of the prop for documentation purposes, not the actual story implementation. Can we talk through this at or after standup?

File: src/components/TextArea/__snapshots__/TextArea.test.tsx.snap [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      Testing TextArea
+      *
+    </label>
+    <textarea
+      aria-describedby=""
+      class="usa-textarea"
+      id="input-type-textarea"
+      name="testing-textarea"
+      placeholder="Can you place this?"
+      required=""
```
Comment (jdinh8124) [inline src/components/TextArea/__snapshots__/TextArea.test.tsx.snap:42]: I was noticing this on my components, booleans were rendering as strings, was wondering if you've encountered this before

**Review by jdinh8124 — APPROVED** [review]

**Pull request merged**